### DISCUSSION TITLE
Fabric canvas is not rendering if its inside a div that being hidden by ngIf condition

### DISCUSSION THREAD

**Discussion by Huyjoon1995** [fabricjs/fabric.js#6899] [Q&A]
Hi,
I'm currently using fabric in angular project. I want to display a canvas on click of a button so I hided canvas inside a div and only show it if the button being clicked. But when the button being clicked the canvas does not show up on the browser at all. Below is the link to see the problem:
https://codesandbox.io/s/quizzical-mclaren-ivnoz?file=/src/app/app.component.html
Any suggestion of how to fix this issue ?

**Comment by asturur** [comment]
I have zero expertise with angular, i do not know the syntax.
Does that work with showing another div?

**Comment by Huyjoon1995** [comment]
regular html tag is fine but when i debug this situation only the upper canvas show not the lower canvas. The ngIf is just basically hide the canvas it’s only show if you click a button.
